cluster-autoscaler/cloudprovider/gce/gce_url_test.go

```diff
@@ -23,21 +23,54 @@ import (
 )
 
 func TestParseUrl(t *testing.T) {
+	// Valid 'www' URL for instanceGroups
 	proj, zone, name, err := parseGceUrl("https://www.googleapis.com/compute/v1/projects/mwielgus-proj/zones/us-central1-b/instanceGroups/kubernetes-minion-group", "instanceGroups")
 	assert.Nil(t, err)
 	assert.Equal(t, "mwielgus-proj", proj)
 	assert.Equal(t, "us-central1-b", zone)
 	assert.Equal(t, "kubernetes-minion-group", name)
 
+	// Valid 'www' URL for instances
+	proj, zone, name, err = parseGceUrl("https://www.googleapis.com/compute/v1/projects/mwielgus-proj/zones/us-central1-b/instances/my-instance-1", "instances")
+	assert.Nil(t, err)
+	assert.Equal(t, "mwielgus-proj", proj)
+	assert.Equal(t, "us-central1-b", zone)
+	assert.Equal(t, "my-instance-1", name)
+
+	// Valid 'content' URL for instanceGroups
 	proj, zone, name, err = parseGceUrl("https://content.googleapis.com/compute/v1/projects/mwielgus-proj/zones/us-central1-b/instanceGroups/kubernetes-minion-group", "instanceGroups")
 	assert.Nil(t, err)
 	assert.Equal(t, "mwielgus-proj", proj)
 	assert.Equal(t, "us-central1-b", zone)
 	assert.Equal(t, "kubernetes-minion-group", name)
 
-	proj, zone, name, err = parseGceUrl("www.onet.pl", "instanceGroups")
+	// Valid 'content' URL for instances
+	proj, zone, name, err = parseGceUrl("https://content.googleapis.com/compute/v1/projects/mwielgus-proj/zones/us-central1-f/instances/my-instance-2", "instanceGroups")
+	assert.Nil(t, err)
+	assert.Equal(t, "mwielgus-proj", proj)
+	assert.Equal(t, "us-central1-f", zone)
+	assert.Equal(t, "my-instance-2", name)
+
+	// Don't validate subdomains; test that we accept anything ending in .googleapis.com
+	proj, zone, name, err = parseGceUrl("https://18adjadj391.googleapis.com/compute/v1/projects/mwielgus-proj/zones/us-central1-b/instanceGroups/kubernetes-minion-group", "instanceGroups")
+	assert.Nil(t, err)
+	assert.Equal(t, "mwielgus-proj", proj)
+	assert.Equal(t, "us-central1-b", zone)
+	assert.Equal(t, "kubernetes-minion-group", name)
+
+	// Invalid TLD
+	_, _, _, err = parseGceUrl("https://www.googleapis.net/compute/v1/projects/mwielgus-proj/zones/us-central1-b/instanceGroups/kubernetes-minion-group", "instanceGroups")
+	assert.NotNil(t, err)
+
+	// Empty 'expectedResource'
+	_, _, _, err = parseGceUrl("googleapis.net/compute/v1/projects/mwielgus-proj/zones/us-central1-b/instanceGroups/kubernetes-minion-group", "")
```

Comment:
Also, this only tests for extra error check against empty resource. Can you add a test case for unexpected/wrong resource?